src/components/plan/PlanNavButton.svelte

```diff
@@ -23,14 +29,46 @@
   export let title: string;
 
   let menu: Menu;
+
+  // Add delay functions to handle the navbar button hover
+  let showTimeout: number | undefined;
+  let hideTimeout: number | undefined;
+
+  function delayedShow() {
+    // Clear any existing hide timeout to prevent hiding if we're showing again
+    if (hideTimeout !== undefined) {
+      clearTimeout(hideTimeout);
+    }
+
+    // If no other nav button is currently hovered, apply the delay
+    if (!$isNavHovered) {
+      showTimeout = window.setTimeout(() => {
+        console.log('delayedShow');
+        menu.show();
+        isNavHovered.set(true);
+      }, 500); // Show delay in milliseconds
+    } else {
+      // If another nav button is already hovered, show the menu immediately
+      menu.show();
+    }
+  }
+
+  function delayedHide() {
+    if (showTimeout !== undefined) {
+      clearTimeout(showTimeout);
+    }
+
+    hideTimeout = window.setTimeout(() => {
+      // Check if the mouse is not over any nav button
+      if (!document.querySelector('.nav-button:hover')) {
```

Comment:
Can you check if the state of `isNavHovered` is `true` here instead of doing a dom check?

Reply:
Also I don't think `isNavHovered` has to be a store specifically. I think you might be able to get away with just making it a boolean that you set during the mouse events.

```let isNavHovered: boolean = false```

Reply:
@duranb Tried with that approach and it didn't seem as responsive. Often times you'd be hovering a nav item, move to the next, and there would still be a delay. May not have set it up properly.

Reply:
Removed the dom check and the store. Moving the isNavHovered and delay variables into `<script type='modules'>` section gives all instances of the plan nav button access to it. Much more simple now!